File: ahu_dashboard/lib/models/ahu_log.dart
import 'package:json_annotation/json_annotation.dart';

part 'ahu_log.g.dart';

/// Log message from ESP32 AHU unit
@JsonSerializable()
class AhuLog {
  final int ts;              // Timestamp (millis)
  final String lvl;          // Log level (INFO, WARN, ERROR)
  final String msg;          // Log message

  AhuLog({
    required this.ts,
    required this.lvl,
    required this.msg,
  });

  factory AhuLog.fromJson(Map<String, dynamic> json) =>
      _$AhuLogFromJson(json);

  Map<String, dynamic> toJson() => _$AhuLogToJson(this);

  /// Get formatted timestamp
  String get formattedTime {
    final seconds = ts ~/ 1000;
    final hours = (seconds ~/ 3600) % 24;
    final minutes = (seconds ~/ 60) % 60;
    final secs = seconds % 60;
    return '${hours.toString().padLeft(2, '0')}:${minutes.toString().padLeft(2, '0')}:${secs.toString().padLeft(2, '0')}';
  }
}

File: pi_dashboard/app/models/ahu_log.py
from dataclasses import dataclass


@dataclass
class AhuLog:
    ts: int       # epoch millis or uptime millis
    lvl: str      # "INFO" | "WARN" | "ERROR"
    msg: str

    @property
    def formatted_time(self) -> str:
        secs = (self.ts // 1000) % 86400
        h = secs // 3600
        m = (secs % 3600) // 60
        s = secs % 60
        return f"{h:02d}:{m:02d}:{s:02d}"

    @classmethod
    def from_dict(cls, d: dict) -> "AhuLog":
        return cls(
            ts=int(d.get("ts", 0)),
            lvl=str(d.get("lvl", "INFO")).upper(),
            msg=str(d.get("msg", "")),
        )
